File: C#/Refactoring/refactoring_exercise_final/za/co/entelect/refactoring_final/controller/ImageController.cs
using System;
using refactoring_exercise_final.za.co.entelect.refactoring_final.domain;
using refactoring_exercise_final.za.co.entelect.refactoring_final.service;

namespace refactoring_exercise_final.za.co.entelect.refactoring_final.controller
{
    public class ImageController
    {
        private ImageService imageService = new ImageService();

        public Image FetchImage(String id)
        {
            return imageService.Fetch(id);
        }

        public Image UploadImage(String id, byte[] data)
        {
            Image image = new Image(id, data);
            imageService.Add(image);
            return image;
        }
    }
}

File: Java/refactoring_exercise_final/src/main/java/za/co/entelect/refactoring_final/controller/ImageController.java
package za.co.entelect.refactoring_final.controller;

import za.co.entelect.refactoring_final.domain.Image;
import za.co.entelect.refactoring_final.service.ImageService;

public class ImageController {

    private static final long ACCOUNT_REOPEN_FEE_CENTS = 2000;

    private ImageService imageService = new ImageService();

    public Image fetchImage(String id) {
        return imageService.fetch(id);
    }

    public Image uploadImage(String id, byte[] data){
        Image image = new Image(id, data);
        imageService.add(image);
        return image;
    }
}
